gateway-service/src/main/java/org/zowe/apiml/gateway/config/AdditionalRegistrationConfig.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.gateway.config;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.StandardEnvironment;
+import org.springframework.util.CollectionUtils;
+
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+@Slf4j
+@Configuration
+@RequiredArgsConstructor
+public class AdditionalRegistrationConfig {
+
+    public static final String DISCOVERY_SERVICE_URLS_KEY = "DISCOVERYSERVICEURLS";
+    public static final String GATEWAY_URL_KEY = "GATEWAYURL";
+    public static final String COMMON_PREFIX = "ZWE_CONFIGS_APIML_SERVICE_ADDITIONALREGISTRATION_";
+    public static final String SERVICE_URL_KEY = "SERVICEURL";
+    private static final int EXPECTED_ROUTE_PART_INDEX = 2;
+
+    @Bean
+    public List<AdditionalRegistration> additionalRegistration(StandardEnvironment environment) {
+        List<AdditionalRegistration> additionalRegistrations = extractAdditionalRegistrations(System.getenv());
```

Comment:
Do we support also the application.yaml or just system env?

Reply:
just env,  but we will add yaml later